File: 0275-h-index-ii/Solution.go
package Leetcode

func hIndex(citations []int) int {
	left := 0
	right := len(citations) - 1
	for left <= right {
		mid := left + (right-left)/2
		if citations[mid] < len(citations)-mid {
			left = mid + 1
		} else {
			right = mid - 1
		}
	}
	return len(citations) - left
}

File: 0275-h-index-ii/Solution.cs
public class Solution {
    public int HIndex(int[] citations)
    {
        int left = 0;
        int right = citations.Length - 1;
        while (left <= right)
        {
            int mid = left + (right - left) / 2;
            if (citations[mid] < citations.Length - mid)
            {
                left = mid + 1;
            }
            else
            {
                right = mid - 1;
            }
        }
        return citations.Length - left;
    }
}